backend/models/post.js

```diff
@@ -0,0 +1,19 @@
+const mongoose = require('mongoose');
+
+
+
+// Post Schema
+const postSchema = new mongoose.Schema(
+    {
+        post_id: { auto: true }, // Automatically generated ObjectId for the post
+        song_link: { type: String, required: true }, // Link to the song
+        likes: { type: Number, default: 0 }, // Count of likes
+        comments: [{ type: mongoose.Schema.Types.ObjectId, ref: 'Comment' }], // List of comment IDs
```

Comment:
you might want to change this to the type of the user_id field under the user model